runelite-client/src/main/java/net/runelite/client/plugins/runenergy/RunEnergyOverlay.java

```diff
@@ -0,0 +1,87 @@
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.runenergy;
+
+import java.awt.Dimension;
+import java.awt.Graphics2D;
+import java.awt.Rectangle;
+import javax.inject.Inject;
+import net.runelite.api.Client;
+import net.runelite.api.Point;
+import net.runelite.api.widgets.Widget;
+import net.runelite.api.widgets.WidgetInfo;
+import net.runelite.client.ui.overlay.Overlay;
+import net.runelite.client.ui.overlay.OverlayLayer;
+import net.runelite.client.ui.overlay.OverlayPosition;
+import net.runelite.client.ui.overlay.tooltip.Tooltip;
+import net.runelite.client.ui.overlay.tooltip.TooltipManager;
+
+public class RunEnergyOverlay extends Overlay
+{
+	private final RunEnergyPlugin plugin;
+	private final Client client;
+	private final RunEnergyConfig config;
+	private final TooltipManager tooltipManager;
+
+	@Inject
+	private RunEnergyOverlay(final RunEnergyPlugin plugin, final Client client, final TooltipManager tooltipManager, final RunEnergyConfig config)
+	{
+		this.plugin = plugin;
+		this.client = client;
+		this.tooltipManager = tooltipManager;
+		this.config = config;
+		setPosition(OverlayPosition.DYNAMIC);
+		setLayer(OverlayLayer.ABOVE_WIDGETS);
+	}
+
+	@Override
+	public Dimension render(Graphics2D graphics)
+	{
+		final Widget runOrb = client.getWidget(WidgetInfo.MINIMAP_RUN_ORB);
+		if (runOrb == null)
+		{
+			return null;
+		}
+
+		final Rectangle bounds = runOrb.getBounds();
+		if (bounds.getX() <= 0)
+		{
+			return null;
+		}
+
+		final Point mousePosition = client.getMouseCanvasPosition();
+
+		if (bounds.contains(mousePosition.getX(), mousePosition.getY()))
+		{
+			final String tooltip = "Weight: " + client.getWeight() + " kg</br>" +
+				"Run Time Remaining: " + plugin.getEstimatedRunTimeRemaining(false) + "</br>" +
+				"100% Energy In: " + plugin.getEstimatedRecoverTimeRemaining();
+
+			tooltipManager.add(new Tooltip(tooltip));
+		}
+
+		return new Dimension((int) bounds.getWidth(), (int) bounds.getHeight());
```

Comment:
You can just return bounds.getSize() or simply null as this is not layoutable overlay.

Reply:
Thanks. I've fixed this. 👍